cirq/experiments/cross_entropy_benchmarking.py

```diff
@@ -0,0 +1,347 @@
+            automatically chosen based on the row and column indices of the
+            grid qubits, assuming a two-qubit gate is to be applied between
+            all possible pairs of qubits adjacent to each other. A maximum of 4
+            cycles is needed to enact all two-qubit gates, in general. As an
+            example, if the qubits are located on a 4 by 4 grid. The layers
+            of two-qubit gates would look like:
+
+                    Cycle 1:                            Cycle 2:
+            q00 ── q01    q02 ── q03            q00    q01    q02    q03
+                                                 |      |      |      |
+            q10 ── q11    q12 ── q13            q10    q11    q12    q13
+
+            q20 ── q21    q22 ── q23            q20    q21    q22    q23
+                                                 |      |      |      |
+            q30 ── q31    q32 ── q33            q30    q31    q32    q33
+
+                     Cycle 3:                           Cycle 4:
+            q00    q01 ── q02    q03            q00    q01    q02    q03
+
+            q10    q11 ── q12    q13            q10    q11    q12    q13
+                                                 |      |      |      |
+            q20    q21 ── q22    q23            q20    q21    q22    q23
+
+            q30    q31 ── q32    q33            q30    q31    q32    q33
+
+        use_tetrahedral_group: If False (by default), the single-qubit
+            gates are chosen from X/2 (\pi/2 rotation around the X axis),
+            Y/2 (\pi/2 rotation around the Y axis) and (X + Y)/2 (\pi/2
+            rotation around an axis \pi/4 away from the X on the equator of
+            the Bloch sphere). If True, the single-qubit gates are chosen
+            from the tetrahedral group, which consists of 12 gates constructed
+            from I, \pm X/2, \pm Y/2, X and Y rotations. Refer to Barends et
+            al., Phys. Rev. A 90, 030303(R) for details on the tetrahedral
+            group.
+
+    Returns:
+        A CrossEntropyResult object that stores and plots the result.
+    """
+    simulator = sim.Simulator()
+    num_qubits = len(qubits)
+
+    if interaction_sequence is None:
+        interaction_sequence = _interaction_sequence(qubits)
+
+    # These store the measured and simulated bit-string probabilities from
+    # all trials in two dictionaries. The keys of the dictionaries are the
+    # numbers of cycles. The values are 2D arrays with each row being the
+    # probabilities obtained from a single trial.
+    probs_meas = {n: np.zeros((num_circuits, 2 ** num_qubits)) for n in
+                  num_cycle_range}
+    probs_exp = {n: np.zeros((num_circuits, 2 ** num_qubits)) for n in
+                 num_cycle_range}
+
+    for k in range(num_circuits):
+
+        # Generates one random XEB circuit with max(num_cycle_range) cycles.
+        # Then the first n cycles of the circuit are taken to generate
```

Comment:
Is this safe to do? It doesn't introduce bad correlations into the results?

Reply:
This is what we do in experiment: use the first x cycles of a long circuit as short circuits. This gives the same fitted Pauli error rate as the case where all circuits are randomly chosen.